src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,231 @@
+    IOException, InterruptedException {
+    SalesforceClient.queryObjectID(objectName);
+    String id = SalesforceClient.id;
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    SalesforceClient.queryObject(id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse, targetTable);
+
+  }
+
+  public static boolean validateSalesforceMultiObjectToBQ(String objectName) throws IOException,
+    InterruptedException {
+    String id = SalesforceClient.id;
+    String targetTable = getTableByName();
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    SalesforceClient.queryObject(id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse, targetTable);
+  }
+
+  private static void getBigQueryTableData(String table, List<Object> bigQueryRows) throws IOException,
+    InterruptedException {
+
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    result.iterateAll().forEach(value -> bigQueryRows.add(value.get(0).getValue()));
+  }
+
+  public static TableResult getTableNamesFromDataSet() throws IOException, InterruptedException {
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT table_name FROM `" + projectId + "." + dataset + "`.INFORMATION_SCHEMA.TABLES ";
+
+    return BigQueryClient.getQueryResult(selectQuery);
+  }
+
+  public static String getTableByName() throws IOException, InterruptedException {
+    String tableName = PluginPropertyUtils.pluginProp("bqTargetTable");
+    TableResult tableResult = getTableNamesFromDataSet();
+    Iterable<FieldValueList> rows = tableResult.iterateAll();
+
+    for (FieldValueList row : rows) {
+      FieldValue fieldValue = row.get(0);
+      String currentTableName = fieldValue.getStringValue();
+
+      if (currentTableName.equals(tableName)) {
+        return currentTableName;
+      }
+    }
+
+    return null; // Table not found
+  }
+
+  public static boolean compareSalesforceAndJsonData(List<JsonObject> salesforceData, List<JsonObject> bigQueryData,
+                                                     String tableName) {
+    boolean result = false;
+    if (bigQueryData == null) {
+      Assert.fail("bigQueryData is null");
+      return result;
+    }
+
```

Comment:
You may have to add a edge case to check the row count of both source and sink

Reply:
done